sample_factory/algo/runners/runner.py

```diff
@@ -535,8 +577,13 @@ def connect_components(self):
             self._setup_component_termination(self.stop, batcher)
             self._setup_component_termination(batcher.stop, learner)
 
+            # Heartbeats
+            self._setup_component_heartbeat(batcher)
+            self._setup_component_heartbeat(learner)
+
         for sampler_component in sampler.stoppable_components():
```

Comment:
Function name "stoppable components" is a bit confusing. Maybe let's have two functions "stoppable components" and "heartbeat components"? And internally just return the same list?

Also, abstract sampler itself should probably have heartbeat too. Current sampler object itself runs on main thread but the interface allows samplers that run in a separate process (i,e, you can imagine a synchronous sampler that does inference and rollout on the same process). 

Although this is not very important for now since we don't have any other implementations.